File: leetcode/746.使用最小花费爬楼梯/main.py
from typing import List


class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        len_cost = len(cost)

        dp = [0] * (len_cost + 1)

        for i in range(2, len_cost + 1):
            dp[i] = min(dp[i - 1] + cost[i - 1], dp[i - 2] + cost[i - 2])

        return dp[len_cost]

File: leetcode/746.使用最小花费爬楼梯/main.go
package main

func minCostClimbingStairs(cost []int) int {
	lenCost := len(cost)
	dp := make([]int, lenCost+1)

	for i := 2; i <= lenCost; i++ {
		dp[i] = min(dp[i-1]+cost[i-1], dp[i-2]+cost[i-2])
	}

	return dp[lenCost]
}

func min(a, b int) int {
	if a < b {
		return a
	}

	return b
}
